/*
 * Copyright 2007 Sun Microsystems, Inc.  All rights reserved.
 *
 * Permission is hereby granted, free of charge, to any person obtaining a
 * copy of this software and associated documentation files (the
 * "Software"), to deal in the Software without restriction, including
 * without limitation the rights to use, copy, modify, merge, publish,
 * distribute, and/or sell copies of the Software, and to permit persons
 * to whom the Software is furnished to do so, provided that the above
 * copyright notice(s) and this permission notice appear in all copies of
 * the Software and that both the above copyright notice(s) and this
 * permission notice appear in supporting documentation.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
 * OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
 * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT
 * OF THIRD PARTY RIGHTS. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR
 * HOLDERS INCLUDED IN THIS NOTICE BE LIABLE FOR ANY CLAIM, OR ANY SPECIAL
 * INDIRECT OR CONSEQUENTIAL DAMAGES, OR ANY DAMAGES WHATSOEVER RESULTING
 * FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN ACTION OF CONTRACT,
 * NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF OR IN CONNECTION
 * WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
 *
 * Except as contained in this notice, the name of a copyright holder
 * shall not be used in advertising or otherwise to promote the sale, use
 * or other dealings in this Software without prior written authorization
 * of the copyright holder.
 *
 */

Solaris 9 shipped with IPv6 support based on the X.Org draft standard, which 
used "inetv6" - the final standard renamed it to "inet6", but since we already
shipped, we retain the old name for backwards compatibility.  
(See PSARC 2004/285.)

diff -urp -x '*~' Xtranssock.c Xtranssock.c
--- Xtranssock.c	2006-07-11 11:06:40.000000000 -0700
+++ Xtranssock.c	2006-08-24 15:13:33.433126000 -0700
@@ -197,6 +197,9 @@ static Sockettrans2dev Sockettrans2devta
     {"tcp",AF_INET6,SOCK_STREAM,SOCK_DGRAM,0},
     {"tcp",AF_INET,SOCK_STREAM,SOCK_DGRAM,0}, /* fallback */
     {"inet6",AF_INET6,SOCK_STREAM,SOCK_DGRAM,0},
+#ifdef SUNSOFT /* Backwards compatibility with Solaris 9 */
+    {"inetv6",AF_INET6,SOCK_STREAM,SOCK_DGRAM,0},
+#endif   
 #endif
 #endif /* TCPCONN */
 #ifdef UNIXCONN
